.github/workflows/pull-requests.yml

```diff
@@ -50,7 +55,17 @@ jobs:
         run: yarn
       - name: Run tests
         run: yarn test
-
+  
+  Formatting:
+    name: Check formatting
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      - name: Install Dependencies
+        run: yarn
+      - name: Check formatting
+        run: yarn format-check
```

Comment:
We use a different methodology for formatting check Talawa-API. Which is better and why? We should be using the better method in both.

Reply:
While the method I am using just checks if the codebase is formatted or not, the method using the published GitHub Action (as in talawa-api) formats 
the code using Prettier and commits any changes files when finished. The point to be noted is that it doesn’t use the version of Prettier specified in your dependencies, meaning that local formatting changes applied by your editor or by running Prettier manually could get changed by the CI build if the versions differed. While this could be resolved by always keeping the version of Prettier in your npm dependencies and the GitHub action in sync, this adds extra work that needs to be done when updating dependencies. 

There is another solution that is just as easy to implement! This solution utilizes the git-auto-commit-action alongside the formatting script already present in our repository.

To setup the GitHub action, all we need to do is install our dependencies, run our format script, and then commit any changes if necessary.

```
name: Format
on:
   pull_request:
      branches: [main]
jobs:
   format:
      runs-on: ubuntu-latest
      steps:
         - uses: actions/checkout@v2
            with:
                ref: ${{ github.head_ref }}
         - uses: actions/setup-node@v2
            with:
                node-version: "14.x"
         - run: yarn
         - run: yarn format
         - name: Commit changes
             uses: stefanzweifel/git-auto-commit-action@v4
             with:
                commit_message: Apply formatting changes
                branch: ${{ github.head_ref }}
```
[Source](https://mskelton.medium.com/auto-formatting-code-using-prettier-and-github-actions-ed458f58b7df#:~:text=Prettier%20is%20a%20very%20popular,ongoing%20debates%20about%20code%20styles)